sklearn/preprocessing/data.py

```diff
@@ -1904,44 +1904,75 @@ def transform(self, X):
                                    self.categorical_features, copy=True)
 
 
+def _get_count_dict_3():
+    """Gets the outer count dictionary."""
+
+    # get_count_dict() is an optimization on key-value access
+    # where we have multi-layer dicts for accessing dict[X][y] instead of
+    # using dict[(X, y)], because using dict[(X, y)] is inefficient
+    # Using external helper methods is also required to avoid a picking
+    # error involving lambda expressions in defaultdict
+    return defaultdict(_get_count_dict_2)
+
+
+def _get_count_dict_2():
+    """Gets the inner count dictionary."""
+    return defaultdict(_get_count_dict_1)
+
+
+def _get_count_dict_1():
+    """Gets the inner count dictionary."""
+    return defaultdict(_get_count_dict_0)
+
+
+def _get_count_dict_0():
+    """Gets the innermost count dictionary."""
+    return defaultdict(int)
```

Comment:
You might want to use a `Counter` here. A difference which may be valuable is that looking up a key in a defaultdict *sets* that key. Looking it up in a `Counter` will return 0 but not expand the dict.

For similar reasons you should not use a `defaultdict` with an active `default_factory` at `transform` time.